rel/value_set_array.go

```diff
@@ -1,7 +1,20 @@
 package rel
 
+import "github.com/arr-ai/frozen"
+
 const ArrayItemAttr = "@item"
 
+// Array represents a sequence of elements.
+type Array interface {
```

Comment:
This interface serves no purpose. You can just define `ArrayEnumerator` as a free function.

Reply:
Wouldn't it be possible in the future to have more functionalities in the `Array` interface? Maybe assigning at an index

Reply:
A more useful addition would be an `Array` struct that implements array-like sets as Go slices. It would be very similar to `String`.